### ISSUE TITLE
outside namespace dependency on consts are not detected

### ISSUE THREAD

**Issue by ricfio** [phparkitect/arkitect#270]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Library Version | 0.2.24 
| PHP version     | 7.1.33

#### Summary

The parser fails to detect const outside namespace as dependency.

#### Current behavior

The dependency with a const is not detected.

#### How to reproduce

```
<?php

namespace Root\Cars;

use AnotherNamespace\CarMake;

class KiaSportage extends AbstractCar
{
    public function __construct()
    {
        parent::__construct(CarMake::KIA, 'Sportage');
    }
}
```

```
$rules = Rule::allClasses()
    ->that(new ResideInOneOfTheseNamespaces('Root\Cars'))
    ->should(new NotHaveDependencyOutsideNamespace('Root\Cars'))
    ->because('we want protect our domain');
```

I haved added a fail test here:
https://github.com/phparkitect/arkitect/pull/269

#### Expected behavior

Dependency detected.

**Comment by AlessandroMinoccheri** [comment]
Thanks @ricfio for your contribution and for adding a test case to understand the problem well!

**Comment by AlessandroMinoccheri** [comment]
Fixed by #271
